src/scitokens/scitokens.py

```diff
@@ -81,12 +84,34 @@ def _derive_algorithm(key):
         returns: Key algorithm if known, otherwise None
         """
 
+        # Issues with PS algorithm are occuring, no way to determine if the scheme is PS or RS based off the key itself, so might have to ask end user what scheme they would like to go with.
         if isinstance(key, rsa.RSAPrivateKey):
-            return "RS256"
+            if key.key_size == 256:
+                return "RS256"
+            elif key.key_size == 384:
+                return "RS384"
+            elif key.key_size == 512:
+                return "RS512"
         elif isinstance(key, ec.EllipticCurvePrivateKey):
-            if key.curve.name == "secp256r1":
+            if key.key_size == 256:
                 return "ES256"
-
+            elif key.key_size == 384:
+                return "ES384"
+            elif key.key_size == 512:
+                return "ES512"
```

Comment:
Is this right? According to https://www.rfc-editor.org/rfc/rfc7518.txt the ES512 algorithm is ECDSA using P-521 and SHA-512 which I think means the key size is 521 not 512.

Reply:
Yup, I believe I made a mistake regarding that. The 521 in P-521 is the keysize. So the comparator should be key.key_size == 521 instead of what it is right now. I will go ahead and change it and repush it so it can run the merge tests again.